gcloud-java-compute/src/main/java/com/google/gcloud/compute/SnapshotDiskConfiguration.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.compute.model.Disk;
+import com.google.common.base.MoreObjects;
+
+import java.util.Objects;
+
+/**
+ * A Google Compute Engine disk configuration to create a disk from a Google Compute Engine
+ * snaoshot.
+ *
+ * @see <a href="https://cloud.google.com/compute/docs/disks/">Block Storage</a>
+ */
+public class SnapshotDiskConfiguration extends DiskConfiguration {
+
+  private static final long serialVersionUID = -1996055058706221049L;
+
+  private final SnapshotId sourceSnapshot;
+  private final String sourceSnapshotId;
+
+  /**
+   * A builder for {@code SnapshotDiskConfiguration} objects.
+   */
+  public static class Builder
+      extends DiskConfiguration.Builder<SnapshotDiskConfiguration, Builder> {
+
+    private SnapshotId sourceSnapshot;
+    private String sourceSnapshotId;
+
+    private Builder() {
+      super(Type.SNAPSHOT);
+    }
+
+    private Builder(SnapshotDiskConfiguration snapshotDiskInfo) {
```

Comment:
I feel like the name containing "DiskInfo" is a bit misleading; maybe change to "snapshotDiskConfiguration" or just "configuration"?